### PR TITLE
[eclipse/xtext#1777] ported xtend code to java

### PR DESCRIPTION
**Pull request by cdietrich**
[eclipse/xtext#1777] ported xtend code to java
Signed-off-by: Christian Dietrich <christian.dietrich@itemis.de>

### CODE REVIEW

File: org.eclipse.xtext.web/src/main/java/org/eclipse/xtext/web/server/model/DocumentSynchronizer.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+	private final AtomicInteger waitingPriorityJobs = new AtomicInteger();
+
+	@Inject
+	private OperationCanceledManager operationCanceledManager;
+
+	@Accessors
+	private volatile boolean canceled;
+
+	public Boolean acquireLock(boolean priority) {
```
Comment (cdietrich) [inline org.eclipse.xtext.web/src/main/java/org/eclipse/xtext/web/server/model/DocumentSynchronizer.java:39]: API from hell

**Comment by cdietrich** [comment]
@szarnekow @tivervac please review

File: org.eclipse.xtext.web/src/main/java/org/eclipse/xtext/web/server/model/DocumentSynchronizer.java (hunk 1 above)
Comment (szarnekow) [reply org.eclipse.xtext.web/src/main/java/org/eclipse/xtext/web/server/model/DocumentSynchronizer.java:39]: (tm)

File: org.eclipse.xtext.web/src/main/java/org/eclipse/xtext/web/server/model/DocumentSynchronizer.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+import org.eclipse.xtend.lib.annotations.Accessors;
+import org.eclipse.xtext.service.OperationCanceledManager;
+import org.eclipse.xtext.util.CancelIndicator;
+import org.eclipse.xtext.xbase.lib.Exceptions;
+
+import com.google.inject.Inject;
+
+/**
+ * Synchronizer object used by the {@link XtextWebDocumentAccess} for managing
```
Comment (szarnekow) [inline org.eclipse.xtext.web/src/main/java/org/eclipse/xtext/web/server/model/DocumentSynchronizer.java:22]: why oh why

File: org.eclipse.xtext.web/src/main/java/org/eclipse/xtext/web/server/model/XtextWebDocument.java [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+	public <T extends IServiceResult> T getCachedServiceResult(AbstractCachedService<T> service,
+			CancelIndicator cancelIndicator, boolean logCacheMiss) {
+		IServiceResult ca = this.cachedServiceResults.get(service.getClass());
+		if (ca != null) {
+			@SuppressWarnings("unchecked")
+			T result = (T) ca;
+			return result;
+		} else {
+			if (logCacheMiss) {
+				XtextWebDocument.LOG.trace("Cache miss for " + service.getClass().getSimpleName());
```
Comment (szarnekow) [inline org.eclipse.xtext.web/src/main/java/org/eclipse/xtext/web/server/model/XtextWebDocument.java:66]: Unnecessary static access

**Pull request merged**